common/forks/forks.go

```diff
@@ -0,0 +1,45 @@
+package forks
+
+import (
+	"math/big"
+	"sort"
+
+	"github.com/scroll-tech/go-ethereum/params"
+)
+
+// CollectSortedForkHeights returns a sorted set of block numbers that one or more forks are activated on
+func CollectSortedForkHeights(config *params.ChainConfig) ([]uint64, map[uint64]bool) {
+	forkHeightsMap := make(map[uint64]bool)
+	for _, fork := range []*big.Int{
+		config.HomesteadBlock,
+		config.DAOForkBlock,
+		config.EIP150Block,
+		config.EIP155Block,
+		config.EIP158Block,
+		config.ByzantiumBlock,
+		config.ConstantinopleBlock,
+		config.PetersburgBlock,
+		config.IstanbulBlock,
+		config.MuirGlacierBlock,
+		config.BerlinBlock,
+		config.LondonBlock,
+		config.ArrowGlacierBlock,
+		config.ArchimedesBlock,
```

Comment:
Shall we define scroll's own hard fork block number? I think the ethereum hard fork block number isn't fit for us

Reply:
I believe that is out-of-scope for this PR, but whenever a new fork block configuration is added to `params.ChainConfig`, it should be added to this list as well.

Reply:
make sense